pkg/operator/observe_config.go

```diff
@@ -92,63 +87,112 @@ func (c ConfigObserver) sync() error {
 	// don't worry about errors
 	currentConfig := map[string]interface{}{}
 	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
-	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
+	observedConfig := map[string]interface{}{}
+	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&observedConfig)
```

Comment:
Reading top to bottom, this just seems wrong.  You have a previous config and you have the observed config you're building up.  Using the previous value means active clearing instead of active adding.  That seems backwards, dangerous, and wrong.

Reply:
For this particular operator, there is nothing to clear as all config keys are required. Is there a situation where the list of required keys might of changed? e.g. during an upgrade of the kube-apiserver?

Reply:
> For this particular operator, there is nothing to clear as all config keys are required. Is there a situation where the list of required keys might of changed? e.g. during an upgrade of the kube-apiserver?

A case where you have optional keys.  A different version of the operator.  Start empty and build up, don't start full and clear.